src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8448,10 +8448,10 @@ private ExpressionSyntax ParseSubExpressionCore(uint precedence)
             {
                 var questionToken = this.EatToken();
 
-                var colonLeft = this.ParseSubExpression(nullCoalescingPrecedence - 1);
+                var colonLeft = this.ParseExpressionCore();
```

Comment:
The purpose of this change is to clearly specify the relative precedence by passing precedence explicitly when parsing subexpressions. I wanted to avoid hard-coding numbers and instead use the computed precedence values. It should have no influence on anything except query expressions, and then only in strict mode. However I will make sure an assignment expression is still accepted between the `?` and the `:`.

I look forward to integrating this with the pattern-matching branch, where the precedence values are represented using an enum instead of raw numbers. Here is the enum:

```cs
        {
            Expression = 0, // Loosest possible precedence, used to accept all expressions
            Assignment,
            Lambda = Assignment, // "The => operator has the same precedence as assignment (=) and is right-associative."
            Ternary,
            Coalescing,
            ConditionalOr,
            ConditionalAnd,
            LogicalOr,
            LogicalXor,
            LogicalAnd,
            Equality,
            Relational,
            Shift,
            Additive,
            Mutiplicative,
            Unary,
            Cast,
            PointerIndirection,
            AddressOf,
            Primary_UNUSED, // Primaries are parsed in an ad-hoc manner.
        }
```